Implement use of linting rule set for Number Verification API

**Problem description**
Implement use of linting rule set for Number Verification API.
what we have to do is well documented in Commonalities:
https://github.com/camaraproject/Commonalities/pull/110
https://github.com/camaraproject/Commonalities/pull/74

**Expected action**
Check our API with linting rule set
provide feedback to commonalities team


**Additional context**
cc: @rartych who asked project volunteer to perform this action.

Offline linting gives the following results:
```
 167:12  warning  camara-properties-descriptions  Property description is missing or empty: "openId.description" property must be truthy     components.securitySchemes.openId
 210:15  warning  camara-properties-descriptions  Property description is missing or empty: "ErrorInfo.description" property must be truthy  components.schemas.ErrorInfo

✖ 2 problems (0 errors, 2 warnings, 0 infos, 0 hints)


number_verification.yaml
  7:1       error    trailing spaces  (trailing-spaces)
  20:1      error    trailing spaces  (trailing-spaces)
  21:24     error    trailing spaces  (trailing-spaces)
  31:1      error    trailing spaces  (trailing-spaces)
  33:1      error    trailing spaces  (trailing-spaces)
  120:11    error    wrong indentation: expected 12 but found 10  (indentation)
  164:11    error    wrong indentation: expected 12 but found 10  (indentation)

```

I propose to clean up the file and then follow the https://github.com/camaraproject/Commonalities/blob/main/documentation/API-linting-Implementation-Guideline.md to implement Github Action in the repository.

--- a/code/API_definitions/number_verification.yaml
+++ b/code/API_definitions/number_verification.yaml
@@ -4,7 +4,7 @@ info:
   description: |
     Service Enabling Network Function API to verify that the provided **mobile phone number** is the one used in the device. It verifies that the user is using a device with the same *mobile phone number* as it is declared.
     It also makes it possible for a Service provider to verify the number itself by returning the phone number associated to the authenticated user's access token.
-    
+
     In this API **phone number** term refers to the mobile phone number
 
     # Introduction
@@ -17,8 +17,8 @@ info:
     # Relevant Definitions and Concepts
 
     - **Network-Based Authentication**: Authentication mechanism based on the identification of the endpoint of a network connection. Network operators know to which user a network resource is assigned at a given moment, for example the mobile phone number associated to a specific mobile network connection.
-  
-    # API Functionality 
+
+    # API Functionality
     This enables a Service Provider (SP) to verify the phone number of the mobile device being used to access their service where the mobile device is accessing the *service provider* over a mobile network (WiFi connections are out of this API scope). This can happen either by getting the comparison result or receiving the phone number of the device that is used, so they can verify it themselves.
 
     # Resources and Operations overview
@@ -28,9 +28,9 @@ info:
 
     # Sequence Diagram
     Number Verification API uses the **standard [OAuth2 Authorization Code grant](https://www.rfc-editor.org/rfc/rfc6749#section-4.1)**. The following diagram will help to clarify the end-to-end process, including previous steps prior to this API call.
-    
+
     ![UML Sequence Diagram](https://raw.githubusercontent.com/camaraproject/NumberVerification/main/documentation/API_documentation/assets/uml_v0.3.jpg)
-    
+
     **Additional details:**
 
     - **(1):** Authentication must be automatic without any user interactions. Authentication methods such as SMS OTP or user/password are incompatible, as the goal is to validate the mobile phone number that is accessing the App. So it is required to be authentication via mobile network and without the user being involved. the use of parameter prompt=none, as described in **[OIDC Connect](https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest)**, ensures no user interaction.
@@ -117,7 +117,7 @@ paths:
           $ref: '#/components/responses/Generic504'
       security:
         - openId:
-          - number-verification:verify
+            - number-verification:verify
   /device-phone-number:
     get:
       tags:
@@ -161,7 +161,7 @@ paths:
           $ref: '#/components/responses/Generic504'
       security:
         - openId:
-          - number-verification:device-phone-number:read
+            - number-verification:device-phone-number:read
 components:
   securitySchemes:
     openId:

